cores/esp8266/umm_malloc/umm_malloc.c

```diff
@@ -550,44 +551,45 @@
 #undef DBG_LOG_FORCE
 
 /* ------------------------------------------------------------------------- */
+#define printf_P(fmt, ...)  do { static const char fstr[] PROGMEM = fmt; char rstr[sizeof(fmt)]; for (size_t i=0; i<sizeof(rstr); i++) rstr[i] = fstr[i]; printf(rstr, ##__VA_ARGS__); } while (0)
```

Comment:
Thanks for the collaborator status.  I'm happy to help out.

I'll refactor the patch this weekend and put in a new one with printf() macro'd instead of the changes inline.